### PR TITLE
{debugger} [GCC-4.9.2] GDB 7.9 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/101b334ef47a61b3902e for a full test report.

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/04296801eea49ae00250 for a full test report.

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/fac0f1b13ff84f039200 for a full test report.

**Comment by wpoely86** [comment]
@rjeschmi you're missing `unxz` on your system.

**Comment by boegel** [comment]
@wpoely86: since that's relatively uncommon, maybe we should add `unxz` as a build dep?

**Comment by wpoely86** [comment]
well, yes and no. It would rather add it to the list of tools that we expect to be there.

**Comment by boegel** [comment]
@wpoely86: http://easybuild.readthedocs.org/en/latest/Installation.html#required-shell-tools

**Comment by rjeschmi** [comment]
Yeah, I'm adding it, but tar could have handled it itself.

**Comment by boegel** [comment]
@rjeschmi: howso?

**Comment by rjeschmi** [comment]
Tar recognizes xz like gz and bz2. It has for a while, but isn't 100%
(although the places where it doesn't would probably require xz compiled
from source)

**Comment by rjeschmi** [comment]
Here it is being piped from the output of unxz to tar. I assume there is some wider compatibility reason for that.

**Comment by rjeschmi** [comment]
nevermind, I'm wrong (I thought it would pull the libraries and not the commandline tools, but it does just run xz)

**Comment by rjeschmi** [comment]
Test report by @rjeschmi
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux centos 6.6, Intel(R) Xeon(R) CPU           E5440  @ 2.83GHz, Python 2.6.6
See https://gist.github.com/a4943cdc66b2238b0475 for a full test report.

**Comment by rjeschmi** [comment]
to save the click:

WARNING: 'makeinfo' is missing on your system.
         You should only need it if you modified a '.texi' file, or
         any other file indirectly affecting the aspect of the manual.
         You might want to install the Texinfo package:
         <http://www.gnu.org/software/texinfo/>
         The spurious makeinfo call might also be the consequence of
         using a buggy 'make' (AIX, DU, IRIX), in which case you might
         want to install GNU make:
         <http://www.gnu.org/software/make/>

**Comment by wpoely86** [comment]
it shouldn't crash on that. Can you look at the configure script? If it doesn't find `makeinfo`, it should try to generate the docs.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/545a307eaacb5498f154 for a full test report.

**Comment by boegel** [comment]
Same issue as @rjeschmi is running into:

```
/tmp/vsc40023/easybuild_build/GDB/7.9/GCC-4.9.2/gdb-7.9/missing: line 81: makeinfo: command not found
WARNING: 'makeinfo' is missing on your system.
```

Other GDB's were fine though...

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/852b4e0d83a4df873751 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/5ca39331f0b89884ef85 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/c961ba4baa3653e75fa4 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/72634f5e29a8a1f59021 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/08df5f05747353400445 for a full test report.

**Comment by wpoely86** [comment]
@boegel @rjeschmi please test again, patched it out.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/09319e47ba7c2d7a18e5 for a full test report.

File: easybuild/easyconfigs/g/GDB/GDB-7.9-missing-makeinfo.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Don't let the make fail when makeinfo is not found
+diff -ur gdb-7.9.orig/missing gdb-7.9/missing
+--- gdb-7.9.orig/missing	2015-02-19 12:58:08.000000000 +0100
++++ gdb-7.9/missing	2015-03-11 14:22:52.000000000 +0100
+@@ -204,7 +204,7 @@
+ 
+ # Propagate the correct exit status (expected to be 127 for a program
+ # not found, 63 for a program that failed due to version mismatch).
+-exit $st
++exit 0
```
Comment (boegel) [inline easybuild/easyconfigs/g/GDB/GDB-7.9-missing-makeinfo.patch]: hmm, this looks kind of a generic change?

you're basically telling the GDB build procedure here that `makeinfo` *is* there? or possible that all tools it checks for are there?

Comment (wpoely86) [inline easybuild/easyconfigs/g/GDB/GDB-7.9-missing-makeinfo.patch]: yes. This is the way they did it in 7.8.
It's just a bad build system. They should rebuild the docs when `makeinfo` is not there.

I've filled a bug about it https://sourceware.org/bugzilla/show_bug.cgi?id=18113

Comment (boegel) [inline easybuild/easyconfigs/g/GDB/GDB-7.9-missing-makeinfo.patch]: please include a reference in the patch comments to where you have find this fix + the bug ticket for reporting this (i.e. the URL you pasted here)

File: easybuild/easyconfigs/g/GDB/GDB-7.9-missing-makeinfo.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Don't let the make fail when makeinfo is not found.
+# The missing script is supposed to handle missing deps for the build,
+# but by return the actual exit code, the build will still fail.
```
Comment (boegel) [inline easybuild/easyconfigs/g/GDB/GDB-7.9-missing-makeinfo.patch]: return*ing*

**Comment by boegel** [comment]
Going in, thanks @wpoely86!

This is a nice example for showing off that testing by multiple people matters. :-)

**Pull request merged**